02/app.js

```diff
@@ -1,32 +1,48 @@
 document.addEventListener('DOMContentLoaded', init);
 
 function init() {
-    const divList = document.querySelectorAll('div');
-    
-    setBorderColorAsync(divList[0], 'red', function() {
-        setBorderColorAsync(divList[1], 'blue', function() {
-            setBorderColorAsync(divList[2], 'green', function() {
-                console.log('finish');
-            });
-        });
-    });
+	const divList = document.querySelectorAll('div');
 
+	setBorderColorAsyncPromise(divList[0], 'red', function () {
+		setBorderColorAsyncPromise(divList[1], 'blue', function () {
+			setBorderColorAsyncPromise(divList[2], 'green', function () {
+				console.log('finish');
+			});
+		});
+	});
 }
 
-function setBorderColorAsync(element, color, callback) {
-    if(element && element instanceof HTMLElement) {
-        // sprawdzam czy parametr jest elementem DOM, więcej:
-        // https://stackoverflow.com/questions/384286/javascript-isdom-how-do-you-check-if-a-javascript-object-is-a-dom-object
-        
-        if(callback && typeof callback === 'function') {
-            setTimeout(() => {
-                element.style.border = `3px solid ${color}`;
-                callback();
-            }, Math.random() * 3000);
-        } else {
-            alert('Parametr ~callback~ mus być funkcją');
-        }
-    } else {
-        alert('Paremetr ~element~ musi być prawidłowym elementem DOM');
-    }
-}
\ No newline at end of file
+// function setBorderColorAsync(element, color, callback) {
+// 	if (element && element instanceof HTMLElement) {
+// 		// sprawdzam czy parametr jest elementem DOM, więcej:
+// 		// https://stackoverflow.com/questions/384286/javascript-isdom-how-do-you-check-if-a-javascript-object-is-a-dom-object
+
+// 		if (callback && typeof callback === 'function') {
+// 			setTimeout(() => {
+// 				element.style.border = `3px solid ${color}`;
+// 				callback();
+// 			}, Math.random() * 3000);
+// 		} else {
+// 			alert('Parametr ~callback~ mus być funkcją');
+// 		}
+// 	} else {
+// 		alert('Paremetr ~element~ musi być prawidłowym elementem DOM');
+// 	}
+// }
+function setBorderColorAsyncPromise(element, color, callback) {
```

Comment:
Skoro mamy już obietnicę to callback nie jest już potrzebny.
Wtedy też trzeba zmienić sposób wywołania - w poprzednim zadaniu to zrobiłeś ok :P
Rozumiem, że tutaj już nie chciałeś robić druga raz tego samego?